pkg/nodeconfig/nodeconfig.go

```diff
@@ -148,7 +172,12 @@ func NewNodeConfig(clientset *kubernetes.Clientset, clusterServiceCIDR, vxlanPor
 		return nil, errors.Wrap(err, "error instantiating Windows instance from VM")
 	}
 
-	return &nodeConfig{k8sclientset: clientset, Windows: win, network: newNetwork(log),
+	infra, err := getClusterInfrastructure()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get cluster infrastructure resource")
+	}
+
+	return &nodeConfig{k8sclientset: clientset, Windows: win, network: newNetwork(log), platformType: infra.Status.PlatformStatus.Type,
```

Comment:
I think there was a miscommunication here, I meant that the platform type can be passed into NewNodeConfig here:
https://github.com/openshift/windows-machine-config-operator/blob/69a4fee1046d880f55d58b907958d8f75b17643a/controllers/controllers.go#L73
and here:
https://github.com/openshift/windows-machine-config-operator/blob/69a4fee1046d880f55d58b907958d8f75b17643a/controllers/controllers.go#L141

As the instanceReconciler struct already has discovered the platformtype:
https://github.com/openshift/windows-machine-config-operator/blob/69a4fee1046d880f55d58b907958d8f75b17643a/controllers/controllers.go#L53-L54

Reply:
Ah, okay, it's even easier. Done!